src/main/java/org/broadinstitute/consent/http/service/DacService.java

```diff
@@ -294,7 +295,9 @@ List<DataAccessRequest> filterDataAccessRequestsByDac(List<DataAccessRequest> do
   }
 
   public Set<Dac> findByDatasetId(List<Integer> datasetIds) {
-    return dacDAO.findDacsForDatasetIds(datasetIds);
+    Set<Dac> returnSet = new HashSet<>();
+    dacDAO.findDacsForDatasetIds(datasetIds).forEach(dac -> returnSet.add(findById(dac.getDacId())));
```

Comment:
```suggestion
    Set.of(dacDAO.findDacsForDatasetIds(datasetIds).forEach(dac -> findById(dac.getDacId())));
```

Can we use this construction if there are no duplicates? Otherwise there are also a few other immutable constructions:
* `Set.copyOf(Arrays.asList(array));`
* `Arrays.stream(array).collect(Collectors.toSet());`

Reply:
Slightly different implementation was needed but it's now immutable.